src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -229,9 +243,10 @@ public void validate() throws AppEngineException {
           "Validation Error: Java App Engine SDK path '" + getJavaAppEngineSdkPath()
               + "' is not valid");
     }
-    if (!getJavaToolsJar().toFile().isFile()) {
+    if (!JAR_LOCATIONS.get(JAVA_TOOLS_JAR).toFile().isFile()) {
```

Comment:
Files.isRegularFile(). You don't need to convert to a java.io.File with toFile here and above.